Theme cleanup, Fixed Linting, Added Bracket Highlighting

vpillinger opened the pull request:
Cleaned up the code surrounding theme changes. Created a MainMenu class for the purpose of consolidating Menu related code. Also made the menu interface look a little more professional.

Note: Code Mirror allows multiple themes at once, and I thought about implementing this in the menu, but decided against implementing it for now.

Issues resolved:
https://github.com/wellsjo/JSON-Splora/issues/1

wellsjo commented on app/system/MainMenu.js:
```diff
@@ -0,0 +1,73 @@
+		Menu.setApplicationMenu(Menu.buildFromTemplate(menuTemplate))
+	}
+
+	/**
```
little spacing issue here

wellsjo commented on app/system/MainMenu.js:
```diff
@@ -0,0 +1,73 @@
+
+class MainMenu {
+
+	// Create the main menu for the given app
```
please use block / jsdoc-style comments for functions

wellsjo commented on index.html:
```diff
@@ -21,6 +21,8 @@
   <script src="node_modules/codemirror/addon/fold/indent-fold.js" type="text/javascript"></script>
   <script src="node_modules/codemirror/mode/javascript/javascript.js" type="text/javascript"></script>
+  <script src="node_modules/codemirror/addon/edit/matchbrackets.js" type="text/javascript"></script>
+
```
extra line here

vpillinger commented:
I updated the title to reflect what is in my branch now. I incorporated those style fixes. I added bracket matching. I also fixed the linter by registering a helper with CodeMirror. Currently, I am just running the parser and returning the errors if any were found. The thing is that the red underline is either going to be 1 character, or the entire document. The parser isn't really a good linter on its own. However, it is certainly better than no linter.

wellsjo commented:
Awesome! Checking this out now.  We can fix the colors later 😄

wellsjo commented:
I fixed a few style things and changed `MainMenu` to use class syntax. Merging for now, the rest can be added later!

MaanooAk commented on app/system/MainMenu.js:
```diff
@@ -0,0 +1,84 @@
+    this.app = app
+
+    // Add theme sub menu
+    menuTemplate[2].submenu[7].submenu = this.createThemeSubMenu()
```
That's not even running!
The `menuTemplate[2]` must be `menuTemplate[1]`.

MaanooAk commented on app/system/MainMenu.js:
```diff
@@ -0,0 +1,84 @@
+      theme: 'neat',
+      type: 'radio',
+      click: (menuItem, browserWindow, event) => {
+        self.themeClicked(menuItem, browserWindow, event)
```
Here too, `self` to `this`

MaanooAk commented on index.html:
```diff
@@ -79,16 +80,10 @@
-    const {Menu} = remote
-    const menuTemplate = require('./app/system/menu')
-    Menu.setApplicationMenu(Menu.buildFromTemplate(menuTemplate))
+    const app = new App()
```
You renamed `APP` to `app`, can you rename also the refs at `menuTemplate.js` in format and minify

MaanooAk commented:
If don't want I could fix these issues in a separate pull request

wellsjo commented:
@MaanooAk hey, sorry that was really sloppy, I'm fixing in a separate pr
